lib/ansiblelint/__main__.py

```diff
@@ -56,6 +59,20 @@ def initialize_logger(level: int = 0) -> None:
     _logger.debug("Logging initialized to level %s", logging_level)
 
 
+def choose_formatter_factory(
+    options_list: "Namespace"
+) -> Type[formatters.BaseFormatter]:
+    """Select an output formatter based on the incoming command line arguments."""
+    r: Type[formatters.BaseFormatter] = formatters.Formatter
+    if options_list.quiet:
+        r = formatters.QuietFormatter
+    elif options_list.parseable:
+        r = formatters.ParseableFormatter
+    elif options_list.parseable_severity:
+        r = formatters.ParseableSeverityFormatter
+    return r
+
+
 def main() -> int:
     """Linter CLI entry point."""
```

Comment:
@dpendolino flake still thinks that main is too complex: https://github.com/ansible/ansible-lint/pull/931/checks?check_run_id=991487868#step:10:76

Reply:
Removing the complexity whitelisting needs multiple merges in order to reduce it from 13 to 10.